docs/tables/aws_s3_directory_bucket.md

```diff
@@ -0,0 +1,343 @@
+
+```sql+postgres
+select
+  name,
+  region,
+  jsonb_array_elements(policy -> 'Statement') ->> 'Sid' as policy_statement_id,
+  jsonb_array_elements(policy -> 'Statement') ->> 'Effect' as policy_effect,
+  jsonb_array_elements(policy -> 'Statement') -> 'Action' as policy_actions,
+  jsonb_array_elements(policy -> 'Statement') -> 'Resource' as policy_resources
+from
+  aws_s3_directory_bucket
+where
+  policy is not null;
+```
+
+```sql+sqlite
+select
+  name,
+  region,
+  json_extract(value, '$.Sid') as policy_statement_id,
+  json_extract(value, '$.Effect') as policy_effect,
+  json_extract(value, '$.Action') as policy_actions,
+  json_extract(value, '$.Resource') as policy_resources
+from
+  aws_s3_directory_bucket,
+  json_each(policy, '$.Statement')
+where
+  policy is not null;
+```
+
+### Comprehensive directory bucket analysis
+Extract and analyze all configuration details from directory buckets in a single query.
+
+```sql+postgres
+with bucket_configs as (
+  select
+    name,
+    region,
+    jsonb_array_elements(lifecycle_rules) as lifecycle_rule,
+    jsonb_array_elements(server_side_encryption_configuration) as encryption_config,
+    jsonb_array_elements(policy -> 'Statement') as policy_statement
+  from
+    aws_s3_directory_bucket
+  where
+    lifecycle_rules is not null
+    and server_side_encryption_configuration is not null
+    and policy is not null
+)
+select
+  name,
+  region,
+  lifecycle_rule ->> 'ID' as lifecycle_rule_id,
+  lifecycle_rule ->> 'Status' as lifecycle_status,
+  lifecycle_rule -> 'Expiration' ->> 'Days' as expiration_days,
+  encryption_config ->> 'BucketKeyEnabled' as bucket_key_enabled,
+  encryption_config -> 'ApplyServerSideEncryptionByDefault' ->> 'SSEAlgorithm' as encryption_algorithm,
+  policy_statement ->> 'Sid' as policy_statement_id,
+  policy_statement ->> 'Effect' as policy_effect
+from
+  bucket_configs;
+```
+
+```sql+sqlite
+with bucket_configs as (
+  select
+    name,
+    region,
+    json_extract(lifecycle_rule.value, '$.ID') as lifecycle_rule_id,
+    json_extract(lifecycle_rule.value, '$.Status') as lifecycle_status,
+    json_extract(lifecycle_rule.value, '$.Expiration.Days') as expiration_days,
+    json_extract(encryption_config.value, '$.BucketKeyEnabled') as bucket_key_enabled,
+    json_extract(encryption_config.value, '$.ApplyServerSideEncryptionByDefault.SSEAlgorithm') as encryption_algorithm,
+    json_extract(policy_statement.value, '$.Sid') as policy_statement_id,
+    json_extract(policy_statement.value, '$.Effect') as policy_effect
+  from
+    aws_s3_directory_bucket,
+    json_each(lifecycle_rules) as lifecycle_rule,
+    json_each(server_side_encryption_configuration) as encryption_config,
+    json_each(policy, '$.Statement') as policy_statement
+  where
+    lifecycle_rules is not null
+    and server_side_encryption_configuration is not null
+    and policy is not null
+)
+select * from bucket_configs;
```

Comment:
Should this be `select *` or should we get specific fields as mentioned in the CTE?

Reply:
Updated the SQLite query.